app/main.py

```diff
@@ -16,4 +16,17 @@
 }
 collection_of_coins = {1, 2, 25}
 
-# write your code here
+
+def sort_variables_by_mutability(*args) -> dict:
+    sorted_variables = {"mutable": [], "immutable": []}
+
+    for arg in args:
+        if isinstance(arg, (list, set, dict)):
+            sorted_variables["mutable"].append(arg)
+        else:
+            sorted_variables["immutable"].append(arg)
+
+    return sorted_variables
+
+
+sorted_variables = sort_variables_by_mutability(lucky_number, pi, one_is_a_prime_number, name, my_favourite_films, profile_info, marks, collection_of_coins)
```

Comment:
The `sort_variables_by_mutability` function is called with a number of variables as arguments. This is fine, but it would be more flexible to allow the function to accept any iterable as an argument, rather than a fixed number of arguments. This way, you could pass in a list or tuple of variables, which would be more scalable if you have a large number of variables to sort.